File: app/src/main/java/com/upadhyay/hemant/NoteEx/database/NotesDatabase.java
package com.upadhyay.hemant.NoteEx.database;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.upadhyay.hemant.NoteEx.dao.NoteDao;
import com.upadhyay.hemant.NoteEx.entities.Note;

@Database(entities = Note.class, version = 1, exportSchema = false)
public abstract class NotesDatabase extends RoomDatabase {

    private static NotesDatabase notesDatabase;

    public static synchronized NotesDatabase getNotesDatabase(Context context){
        if(notesDatabase == null){
            notesDatabase = Room.databaseBuilder(
                    context,
                    NotesDatabase.class,
                    "notes_database"
            ).build();
        }
        return notesDatabase;
    }

    public abstract NoteDao noteDao();

}

File: app/src/main/java/com/upadhyay/hemant/noteEx/data/database/NotesDatabase.kt
package com.upadhyay.hemant.noteEx.data.database

import androidx.room.Database
import androidx.room.RoomDatabase
import com.upadhyay.hemant.noteEx.data.dao.NoteDao
import com.upadhyay.hemant.noteEx.data.entities.NoteEntity

@Database(entities = [NoteEntity::class], version = 1, exportSchema = false)
abstract class NotesDatabase : RoomDatabase() {

    abstract fun noteDao(): NoteDao?

}